### PR TITLE
[DF-3554] - Html remove scripts from text

### PR DESCRIPTION
**Pull request by jmcadams-r7**
## Proposed Changes

This PR adds the "Remove Scripts" option to the Text action in the HTML plugin 

## Validation: 
```
makermt-mbp-5357:html jmcadams$ make validate
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Colors.mk ../tools/Makefiles/Helpers.mk
--
[*] Running validators
WARNING: You are using pip version 19.2.1, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator SpecPropertiesValidator
[*] Executing validator FilesValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator HelpValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator DockerValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator ExceptionValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 6313.369ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
./komand_html/actions/validate/action.py:7:3: E111 indentation is not a multiple of four
./komand_html/actions/validate/action.py:8:7: E111 indentation is not a multiple of four
./komand_html/actions/validate/action.py:14:3: E111 indentation is not a multiple of four
./komand_html/actions/validate/action.py:21:7: E111 indentation is not a multiple of four
./komand_html/actions/validate/action.py:22:7: E111 indentation is not a multiple of four
./komand_html/actions/validate/action.py:23:7: E111 indentation is not a multiple of four
./komand_html/actions/validate/action.py:25:17: E201 whitespace after '{'
./komand_html/actions/validate/action.py:25:36: E202 whitespace before '}'
./komand_html/actions/validate/action.py:26:7: E111 indentation is not a multiple of four
./komand_html/actions/validate/action.py:27:7: E111 indentation is not a multiple of four
./komand_html/actions/validate/action.py:28:5: F841 local variable 'e' is assigned to but never used
./komand_html/actions/validate/action.py:31:3: E111 indentation is not a multiple of four
./komand_html/actions/html5/action.py:10:3: E111 indentation is not a multiple of four
./komand_html/actions/html5/action.py:11:7: E111 indentation is not a multiple of four
./komand_html/actions/html5/action.py:17:3: E111 indentation is not a multiple of four
./komand_html/actions/html5/action.py:18:24: W605 invalid escape sequence '\/'
./komand_html/actions/html5/action.py:18:27: W605 invalid escape sequence '\w'
./komand_html/actions/html5/action.py:18:32: W605 invalid escape sequence '\s'
./komand_html/actions/html5/action.py:18:35: W605 invalid escape sequence '\w'
./komand_html/actions/html5/action.py:18:39: W605 invalid escape sequence '\s'
./komand_html/actions/html5/action.py:18:43: W605 invalid escape sequence '\s'
./komand_html/actions/html5/action.py:18:69: W605 invalid escape sequence '\s'
./komand_html/actions/html5/action.py:18:78: W605 invalid escape sequence '\s'
./komand_html/actions/html5/action.py:18:82: W605 invalid escape sequence '\s'
./komand_html/actions/html5/action.py:18:86: W605 invalid escape sequence '\/'
./komand_html/actions/html5/action.py:28:5: E722 do not use bare 'except'
./komand_html/actions/html5/action.py:31:3: E111 indentation is not a multiple of four
./komand_html/actions/markdown/action.py:10:3: E111 indentation is not a multiple of four
./komand_html/actions/markdown/action.py:11:7: E111 indentation is not a multiple of four
./komand_html/actions/markdown/action.py:17:3: E111 indentation is not a multiple of four
./komand_html/actions/markdown/action.py:18:24: W605 invalid escape sequence '\/'
./komand_html/actions/markdown/action.py:18:27: W605 invalid escape sequence '\w'
./komand_html/actions/markdown/action.py:18:32: W605 invalid escape sequence '\s'
./komand_html/actions/markdown/action.py:18:35: W605 invalid escape sequence '\w'
./komand_html/actions/markdown/action.py:18:39: W605 invalid escape sequence '\s'
./komand_html/actions/markdown/action.py:18:43: W605 invalid escape sequence '\s'
./komand_html/actions/markdown/action.py:18:69: W605 invalid escape sequence '\s'
./komand_html/actions/markdown/action.py:18:78: W605 invalid escape sequence '\s'
./komand_html/actions/markdown/action.py:18:82: W605 invalid escape sequence '\s'
./komand_html/actions/markdown/action.py:18:86: W605 invalid escape sequence '\/'
./komand_html/actions/markdown/action.py:27:5: E722 do not use bare 'except'
./komand_html/actions/markdown/action.py:30:3: E111 indentation is not a multiple of four
./komand_html/actions/pdf/action.py:6:1: F401 'os' imported but unused
./komand_html/actions/pdf/action.py:12:3: E111 indentation is not a multiple of four
./komand_html/actions/pdf/action.py:13:7: E111 indentation is not a multiple of four
./komand_html/actions/pdf/action.py:19:3: E111 indentation is not a multiple of four
./komand_html/actions/pdf/action.py:21:24: W605 invalid escape sequence '\/'
./komand_html/actions/pdf/action.py:21:27: W605 invalid escape sequence '\w'
./komand_html/actions/pdf/action.py:21:32: W605 invalid escape sequence '\s'
./komand_html/actions/pdf/action.py:21:35: W605 invalid escape sequence '\w'
./komand_html/actions/pdf/action.py:21:39: W605 invalid escape sequence '\s'
./komand_html/actions/pdf/action.py:21:43: W605 invalid escape sequence '\s'
./komand_html/actions/pdf/action.py:21:69: W605 invalid escape sequence '\s'
./komand_html/actions/pdf/action.py:21:78: W605 invalid escape sequence '\s'
./komand_html/actions/pdf/action.py:21:82: W605 invalid escape sequence '\s'
./komand_html/actions/pdf/action.py:21:86: W605 invalid escape sequence '\/'
./komand_html/actions/pdf/action.py:29:13: E265 block comment should start with '# '
./komand_html/actions/pdf/action.py:31:5: E722 do not use bare 'except'
./komand_html/actions/pdf/action.py:34:3: E111 indentation is not a multiple of four
./komand_html/actions/text/action.py:6:1: E302 expected 2 blank lines, found 1
./komand_html/actions/text/action.py:18:20: E261 at least two spaces before inline comment
./komand_html/actions/docx/action.py:11:3: E111 indentation is not a multiple of four
./komand_html/actions/docx/action.py:12:7: E111 indentation is not a multiple of four
./komand_html/actions/docx/action.py:18:3: E111 indentation is not a multiple of four
./komand_html/actions/docx/action.py:20:24: W605 invalid escape sequence '\/'
./komand_html/actions/docx/action.py:20:27: W605 invalid escape sequence '\w'
./komand_html/actions/docx/action.py:20:32: W605 invalid escape sequence '\s'
./komand_html/actions/docx/action.py:20:35: W605 invalid escape sequence '\w'
./komand_html/actions/docx/action.py:20:39: W605 invalid escape sequence '\s'
./komand_html/actions/docx/action.py:20:43: W605 invalid escape sequence '\s'
./komand_html/actions/docx/action.py:20:69: W605 invalid escape sequence '\s'
./komand_html/actions/docx/action.py:20:78: W605 invalid escape sequence '\s'
./komand_html/actions/docx/action.py:20:82: W605 invalid escape sequence '\s'
./komand_html/actions/docx/action.py:20:86: W605 invalid escape sequence '\/'
./komand_html/actions/docx/action.py:28:13: E265 block comment should start with '# '
./komand_html/actions/docx/action.py:30:5: E722 do not use bare 'except'
./komand_html/actions/docx/action.py:33:3: E111 indentation is not a multiple of four
./komand_html/actions/epub/action.py:6:1: F401 'os' imported but unused
./komand_html/actions/epub/action.py:12:3: E111 indentation is not a multiple of four
./komand_html/actions/epub/action.py:13:7: E111 indentation is not a multiple of four
./komand_html/actions/epub/action.py:19:3: E111 indentation is not a multiple of four
./komand_html/actions/epub/action.py:21:24: W605 invalid escape sequence '\/'
./komand_html/actions/epub/action.py:21:27: W605 invalid escape sequence '\w'
./komand_html/actions/epub/action.py:21:32: W605 invalid escape sequence '\s'
./komand_html/actions/epub/action.py:21:35: W605 invalid escape sequence '\w'
./komand_html/actions/epub/action.py:21:39: W605 invalid escape sequence '\s'
./komand_html/actions/epub/action.py:21:43: W605 invalid escape sequence '\s'
./komand_html/actions/epub/action.py:21:69: W605 invalid escape sequence '\s'
./komand_html/actions/epub/action.py:21:78: W605 invalid escape sequence '\s'
./komand_html/actions/epub/action.py:21:82: W605 invalid escape sequence '\s'
./komand_html/actions/epub/action.py:21:86: W605 invalid escape sequence '\/'
./komand_html/actions/epub/action.py:29:17: E117 over-indented (comment)
./komand_html/actions/epub/action.py:29:17: E265 block comment should start with '# '
./komand_html/actions/epub/action.py:30:17: E117 over-indented
./komand_html/actions/epub/action.py:31:5: E722 do not use bare 'except'
./komand_html/actions/epub/action.py:34:3: E111 indentation is not a multiple of four
[FAIL] Fails flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 2.7.15
Run started:2019-11-07 20:12:29.288833

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 622
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0
                Low: 0
                Medium: 0
                High: 0
        Total issues (by confidence):
                Undefined: 0
                Low: 0
                Medium: 0
                High: 0
Files skipped (0):
[SUCCESS] Passes bandit security checks
```

## Tests: 
[html_tests.txt](https://github.com/rapid7/insightconnect-plugins/files/3821473/html_tests.txt)

## Workflow: 
image

### Artifacts: 
image

### Input: 
image

### Output: 
image

### CODE REVIEW

File: html/komand_html/actions/docx/action.py [hunk 1]
```diff
@@ -20,14 +21,14 @@ def run(self, params={}):
     tags = re.findall(tag_parser, params.get('doc'))
     try:
         if not len(tags):
-            raise Exception('Run: Input Invalid, Input must be of type HTML')
+            raise PluginException(cause='Run: Input Invalid',
```
Comment (gusingh-r7) [inline html/komand_html/actions/docx/action.py]:
```suggestion
            raise PluginException(cause='Run: Input Invalid.',
```

File: html/komand_html/actions/docx/action.py [hunk 2]
```diff
@@ -20,14 +21,14 @@ def run(self, params={}):
...
+                                  assistance='Input must be of type HTML')
```
Comment (gusingh-r7) [inline html/komand_html/actions/docx/action.py]:
```suggestion
                                  assistance='Input must be of type HTML.')
```

Comment (gusingh-r7) [reply html/komand_html/actions/docx/action.py]: Please update all PluginException.

File: html/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM komand/python-3-plugin
+FROM komand/python-pypy3-full-plugin
```
Comment (mrinehart-r7) [inline html/Dockerfile]: Why do you need the full SDK image?

File: html/plugin.spec.yaml [hunk 4]
```diff
@@ -129,6 +129,11 @@ actions:
         description: Document to transform
         type: string
         required: true
+      remove_scripts:
+        title: Remove Scripts
+        description: Remove scripts from doc as well as HTML
```
Comment (mrinehart-r7) [inline html/plugin.spec.yaml]:
```suggestion
        description: Remove non-HTML scripts from the document
```

File: html/komand_html/actions/docx/action.py [hunk 5]
```diff
@@ -20,14 +21,14 @@ def run(self, params={}):
...
     try:
         if not len(tags):
-            raise Exception('Run: Input Invalid, Input must be of type HTML')
+            raise PluginException(cause='Run: Input Invalid.',
+                                  assistance='Input must be of type HTML.')
         pypandoc.convert(params.get('doc'), 'docx', outputfile=temp_file, format='html')
         with open(temp_file, 'rb') as output:
             #Reading the output and sending it in base64
             return {'docx': base64.b64encode(output.read()).decode('utf-8')}
-        os.remove(temp_file)
```
Comment (mrinehart-r7) [inline html/komand_html/actions/docx/action.py]: Did you mean to remove this?

File: html/komand_html/actions/epub/action.py [hunk 6]
```diff
@@ -20,13 +22,12 @@ def run(self, params={}):
...
     try:
         if not len(tags):
-            raise Exception('Run: Input Invalid, Input must be of type HTML')
+            raise PluginException(cause='Run: Input Invalid.',
+                                  assistance='Input must be of type HTML.')
         pypandoc.convert(params.get('doc'), 'epub', outputfile=temp_file, format='html')
         with open(temp_file, 'rb') as output:
                 #Reading the output and sending it in base64
                 return {'epub': base64.b64encode(output.read()).decode('utf-8')}
-        os.remove(temp_file)
```
Comment (mrinehart-r7) [inline html/komand_html/actions/epub/action.py]: Did you mean to remove this?

File: html/komand_html/actions/html5/action.py [hunk 7]
```diff
@@ -18,7 +19,8 @@ def run(self, params={}):
     tags = re.findall(tag_parser, params.get('doc'))
     try:
         if not len(tags):
-            raise Exception('Run: Input Invalid, Input must be of type HTML')
+            raise PluginException(cause='Run: Input Invalid.',
```
Comment (mrinehart-r7) [inline html/komand_html/actions/html5/action.py]:
```suggestion
            raise PluginException(cause='Run: Invalid input.',
```

File: html/komand_html/actions/markdown/action.py [hunk 8]
```diff
@@ -18,7 +19,8 @@ def run(self, params={}):
     tags = re.findall(tag_parser, params.get('doc'))
     try:
         if not len(tags):
-            raise Exception('Run: Input Invalid, Input must be of type HTML')
+            raise PluginException(cause='Run: Input Invalid.',
```
Comment (mrinehart-r7) [inline html/komand_html/actions/markdown/action.py]:
```suggestion
            raise PluginException(cause='Run: Invalid input.',
```

File: html/komand_html/actions/pdf/action.py [hunk 9]
```diff
@@ -20,12 +22,12 @@ def run(self, params={}):
     tags = re.findall(tag_parser, params.get('doc'))
     try:
         if not len(tags):
-            raise Exception('Run: Input Invalid, Input must be of type HTML')
+            raise PluginException(cause='Run: Input Invalid.',
```
Comment (mrinehart-r7) [inline html/komand_html/actions/pdf/action.py]:
```suggestion
            raise PluginException(cause='Run: Invalid input.',
```

File: html/Dockerfile (hunk 3 above)
Comment (jmcadams-r7) [reply html/Dockerfile]: I'm not 100% I do, but it bombed when it was using the older image. I tried building it with slim and condensed and it failed. I stuck with what was there originally.

File: html/komand_html/actions/epub/action.py (hunk 6 above)
Comment (jmcadams-r7) [reply html/komand_html/actions/epub/action.py]: Yes. Unreachable code.

File: html/help.md [hunk 10]
```diff
@@ -120,6 +120,7 @@ This action is used to strip an HTML string of all tags and return only the text
 |Name|Type|Default|Required|Description|Enum|
 |----|----|-------|--------|-----------|----|
 |doc|string|None|True|Document to transform|None|
+|remove_scripts|boolean|None|False|Remove scripts from doc as well as HTML|None|
```
Comment (mrinehart-r7) [inline html/help.md]:
```suggestion
|remove_scripts|boolean|None|False|Remove non-HTML scripts from the document|None|
```

Comment (mrinehart-r7) [reply html/help.md]: Just need to update this

**Comment by mrinehart-r7** [comment]
LGTM 👍

**Pull request merged**